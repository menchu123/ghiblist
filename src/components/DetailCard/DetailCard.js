import "./DetailCard.scss";

const DetailCard = ({ movie }) => {
  return (
    <>
      <section className="movie-detail">
        <img
          className="movie-detail__image"
          src={movie.movie_banner}
          alt={`${movie.title} banner`}
          height="300"
        />
        <section className="movie-detail__info">
          <div className="movie-detail__data">
            <p className="movie-detail__year">{movie.release_date}</p>
            <p className="movie-detail__runtime">{movie.running_time}min</p>
            <p className="movie-detail__score">{movie.rt_score}%</p>
          </div>
        </section>
        <h2 className="movie-detail__title">{movie.title}</h2>
        <p className="movie-detail__director">
          {movie.director} Â· {movie.producer}
        </p>
        <p className="movie-detail__description">{movie.description}</p>
      </section>
    </>
  );
};

export default DetailCard;
