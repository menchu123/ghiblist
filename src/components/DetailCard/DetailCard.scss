@import "../../styles-assets/extends";

.movie-detail {
  width: 573px;
  margin: 20px auto 0;
  padding: 20px;
  border: 1px solid #3f37c9;
  &__image {
    border: 1px solid #3f37c9;
  }
  &__info {
    margin-top: 10px;
  }
  &__data {
    display: flex;
    font-size: 14px;
    width: 200px;
    justify-content: space-between;
    padding-right: 25px;
    margin: 5px;
    font-style: italic;
  }
  &__title {
    margin-bottom: 5px;
  }
  &__director {
    margin-bottom: 30px;
  }
}

.loader {
  @extend %loader;
  margin: 1000px auto;
}

.error {
  width: 100%;
  text-align: center;
  padding-top: 200px;
}

@media only screen and (max-width: 620px) {
  .movie-detail {
    width: 100%;
    margin: 0;
    padding: 0;
    border: none;
    &__image {
      border: none;
      width: 100%;
      height: 300px;
      object-fit: cover;
    }
    &__info,
    &__title,
    &__director,
    &__description {
      margin: 20px;
      padding: 0;
    }
    &__data {
      margin-left: 0;
    }
  }
}
