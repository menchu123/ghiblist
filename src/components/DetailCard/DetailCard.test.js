import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import ReactTestRenderer from "react-test-renderer";
import DetailCard from "./DetailCard";

describe("Given a DetailCard component", () => {
  describe("When it is passed a movie object with the title 'Spirited Away'", () => {
    test("Then it should render a card with the movie's title", () => {
      const expectedName = "Spirited Away";
      const movie = {
        title: expectedName,
        director: "Hayao Miyazaki",
        producer: "Isao Takahata",
        release_date: "1986",
        running_time: "124",
        rt_score: "95",
      };

      render(
        <BrowserRouter>
          <DetailCard movie={movie} />
        </BrowserRouter>
      );

      const movieTitle = screen.getByRole("heading");

      expect(movieTitle.textContent).toBe(expectedName);
    });
  });
  describe("When it is passed a movie object", () => {
    test("Then it should render a card matching the snapshot", () => {
      const movie = {
        title: "Spirited Away",
        director: "Hayao Miyazaki",
        producer: "Isao Takahata",
        release_date: "1986",
        running_time: "124",
        rt_score: "95",
      };

      const detailCard = ReactTestRenderer.create(
        <BrowserRouter>
          <DetailCard movie={movie} />
        </BrowserRouter>
      );
      expect(detailCard.toJSON()).toMatchSnapshot();
    });
  });
});
