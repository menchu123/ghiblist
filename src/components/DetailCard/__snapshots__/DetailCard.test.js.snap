// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a DetailCard component When it is passed a movie object Then it should render a card matching the snapshot 1`] = `
<section
  className="movie-detail"
>
  <img
    alt="Spirited Away banner"
    className="movie-detail__image"
    height="300"
  />
  <section
    className="movie-detail__info"
  >
    <div
      className="movie-detail__data"
    >
      <p
        className="movie-detail__year"
      >
        1986
      </p>
      <p
        className="movie-detail__runtime"
      >
        124
        min
      </p>
      <p
        className="movie-detail__score"
      >
        95
        %
      </p>
    </div>
  </section>
  <h2
    className="movie-detail__title"
  >
    Spirited Away
  </h2>
  <p
    className="movie-detail__director"
  >
    Hayao Miyazaki
     Â· 
    Isao Takahata
  </p>
  <p
    className="movie-detail__description"
  />
</section>
`;
